packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,630 @@
+  }
+  if (apiVersion === "latest" || apiVersion === undefined || !allApiVersions.includes(apiVersion)) {
+    apiVersion = allApiVersions[allApiVersions.length - 1];
+  }
+
+  const exampleDir = apiVersion
+    ? resolvePath(context.examplesDirectory, apiVersion)
+    : resolvePath(context.examplesDirectory);
+  try {
+    if (!(await context.program.host.stat(exampleDir)).isDirectory())
+      return diagnostics.wrap(new Map());
+  } catch (err) {
+    diagnostics.add(
+      createDiagnostic({
+        code: "example-loading",
+        messageId: "noDirectory",
+        format: { directory: exampleDir },
+        target: NoTarget,
+      })
+    );
+    return diagnostics.wrap(new Map());
+  }
+
+  const map = new Map<string, Record<string, LoadedExample>>();
+  const exampleFiles = await context.program.host.readDir(exampleDir);
+  for (const fileName of exampleFiles) {
+    try {
+      const exampleFile = await context.program.host.readFile(resolvePath(exampleDir, fileName));
+      const example = JSON.parse(exampleFile.text);
+      if (!example.operationId || !example.title) {
+        diagnostics.add(
+          createDiagnostic({
+            code: "example-loading",
+            messageId: "noOperationId",
+            format: { filename: fileName },
+            target: NoTarget,
+          })
+        );
+        continue;
+      }
+
+      if (!map.has(example.operationId)) {
+        map.set(example.operationId.toLowerCase(), {});
+      }
+      const examples = map.get(example.operationId.toLowerCase())!;
+
+      if (example.title in examples) {
+        diagnostics.add(
+          createDiagnostic({
+            code: "duplicate-example-file",
+            target: NoTarget,
+            format: {
+              filename: fileName,
+              operationId: example.operationId,
+              title: example.title,
+            },
+          })
+        );
+      }
+
+      examples[example.title] = {
+        relativePath: fileName,
+        file: exampleFile,
+        data: example,
+      };
+    } catch (err) {
+      diagnostics.add(
+        createDiagnostic({
+          code: "example-loading",
+          messageId: "default",
+          format: { filename: fileName, error: err?.toString() ?? "" },
+          target: NoTarget,
+        })
+      );
+    }
+  }
+  return diagnostics.wrap(map);
+}
+
+export async function handleClientExamples(
+  context: TCGCContext,
+  client: SdkClientType<SdkServiceOperation>
+): Promise<[void, readonly Diagnostic[]]> {
+  const diagnostics = createDiagnosticCollector();
+
+  const examples = diagnostics.pipe(await loadExamples(context, client));
+  const methodQueue = [...client.methods];
+  while (methodQueue.length > 0) {
+    const method = methodQueue.pop()!;
+    if (method.kind === "clientaccessor") {
+      methodQueue.push(...method.response.methods);
+    } else {
+      const operationId = resolveOperationId(context.program, method.__raw!).toLowerCase();
```

Comment:
one concern here is whether tcgc need to see `@example` defined in `typespec-autorest`.

Reply:
is it a concern because we can't use decorators from `typespec-autorest`? If so, the `.decorators` PR should resolve this issue, right?

Reply:
yes. that will make it possible to get example value without import the lib. thanks for reminder.

Reply:
i went through all the current spec and it seems there is no actual usage for `@example`. i decided to leave it for now.